src/mprisplayer.cpp

```diff
@@ -93,21 +93,38 @@ qlonglong MprisPlayerPrivate::position() const
 
 void MprisPlayerPrivate::setLoopStatus(const QString &value)
 {
-    bool ok;
-    int enumVal = QMetaEnum::fromType<Mpris::LoopStatus>().keyToValue(value.toUtf8(), &ok);
+    int enumVal;
+    bool ok = true;
 
-    if (!ok) {
-        sendErrorReply(QDBusError::InvalidArgs, QStringLiteral("Invalid loop status"));
-        return;
+    if (value == QLatin1String("None")) {
+        enumVal = Mpris::LoopNone;
+    } else if (value == QLatin1String("Track")) {
+        enumVal = Mpris::LoopTrack;
+    } else if (value == QLatin1String("Playlist")) {
+        enumVal = Mpris::LoopPlaylist;
+    } else {
+        ok = false;
     }
 
-    Q_EMIT parent()->loopStatusRequested(enumVal);
+    if (ok) {
+        Q_EMIT parent()->loopStatusRequested(enumVal);
+    } else {
+        sendErrorReply(QDBusError::InvalidArgs, QStringLiteral("Invalid loop status"));
+    }
 }
 
 QString MprisPlayerPrivate::loopStatus() const
 {
-    const char *strVal = QMetaEnum::fromType<Mpris::LoopStatus>().valueToKey(static_cast<int>(parent()->loopStatus()));
-    return QString::fromLatin1(strVal);
+    switch (parent()->loopStatus()) {
+    case Mpris::LoopNone:
+        return QLatin1String("None");
```

Comment:
Wouldn't QStringLiteral be more efficient here, as the value is used to construct a QString?

Reply:
Well, AFAIU the compiler wouldn't be then able to reuse the c string literals that are now needed on both sides of the mapping. That problem could be avoided by global strings for these, but as this is hardly a hot code path I'm not sure if that's worth it.

Reply:
Yeah, probably not that important.